reconcile/terraform_cloudflare_dns.py

```diff
@@ -0,0 +1,262 @@
+import logging
+import sys
+from typing import (
+    Any,
+    List,
+    Optional,
+)
+
+from reconcile.gql_definitions.terraform_cloudflare_dns import (
+    terraform_cloudflare_zones,
+)
+from reconcile.gql_definitions.terraform_cloudflare_dns.terraform_cloudflare_zones import (
+    CloudflareDnsZoneQueryData,
+    CloudflareDnsZoneV1,
+)
+from reconcile.gql_definitions.terraform_cloudflare_resources import (
+)
+from reconcile.gql_definitions.terraform_cloudflare_resources.terraform_cloudflare_accounts import (
+    AWSAccountV1,
+    CloudflareAccountV1,
+)
+from reconcile.status import ExitCodes
+from reconcile.typed_queries.app_interface_vault_settings import (
+    get_app_interface_vault_settings,
+)
+from reconcile.utils import gql
+from reconcile.utils.defer import defer
+from reconcile.utils.exceptions import SecretIncompleteError
+from reconcile.utils.external_resources import (
+    ExternalResourceSpec,
+)
+from reconcile.utils.secret_reader import (
+    SecretReaderBase,
+    create_secret_reader,
+)
+from reconcile.utils.semver_helper import make_semver
+from reconcile.utils.terraform.config_client import TerraformConfigClientCollection
+from reconcile.utils.terraform_client import TerraformClient
+from reconcile.utils.terrascript.cloudflare_client import (
+    DEFAULT_CLOUDFLARE_ACCOUNT_2FA,
+    DEFAULT_CLOUDFLARE_ACCOUNT_TYPE,
+    CloudflareAccountConfig,
+    TerraformS3BackendConfig,
+    TerrascriptCloudflareClient,
+    create_cloudflare_terrascript,
+)
+
+QONTRACT_INTEGRATION = "terraform_cloudflare_dns"
+QONTRACT_INTEGRATION_VERSION = make_semver(0, 1, 0)
+QONTRACT_TF_PREFIX = "qrtfcfdns"
+
+
+def create_backend_config(
+    secret_reader: SecretReaderBase,
+    aws_acct: AWSAccountV1,
+    cf_acct: CloudflareAccountV1,
+    zone: str,
+) -> TerraformS3BackendConfig:
+    aws_acct_creds = secret_reader.read_all({"path": aws_acct.automation_token.path})
+
+    # default from AWS account file
+    tf_state = aws_acct.terraform_state
+    if tf_state is None:
+        raise ValueError(
+            f"AWS account {aws_acct.name} cannot be used for Cloudflare "
+            f"account {cf_acct.name} because it doesn't define a terraform state "
+        )
+
+    integrations = tf_state.integrations or []
+    for i in integrations or []:
+        name = i.integration
+        if name.replace("-", "_") == QONTRACT_INTEGRATION:
+            # we have to ensure the bucket key(file) is unique across
```

Comment:
We had a bit of discussion about this: https://redhat-internal.slack.com/archives/C0354U32C9M/p1675186237722129

Can we make the comment clearer based on this disussion?